packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -212,6 +213,26 @@ export class SummaryTreeBuilder implements ISummaryTreeWithStats {
 		this.summaryStats = mergeStats(this.summaryStats, summarizeResult.stats);
 	}
 
+	/**
+	 * This function does the following:
+	 * - For handle type result, normalizes the handles and prefixes the given 'prefix' to it.
+	 * - Adds the given summarize result to the summary tree.
+	 * - Adds the stats from the given summarize result to the overall summary stats.
+	 */
+	public addWithStatsAndPrefixHandles(
+		prefix: string,
+		key: string,
+		summarizeResult: ISummarizeResult,
+	): void {
+		if (summarizeResult.summary.type === SummaryType.Handle) {
```

Comment:
if summarizeResult type is the same as type of summaryObject variable in prefixAllSummaryHandles, might be worth to move these 4 lines of code into its own function and call it from both places?